var/spack/repos/builtin/packages/hypre-cmake/package.py

```diff
@@ -99,31 +99,23 @@ def _cached_tests_work_dir(self):
         """The working directory for cached test sources."""
         return join_path(self.test_suite.current_test_cache_dir, self.extra_install_tests)
 
-    def test(self):
+    def test_hypre_cmake(self):
         """Perform smoke test on installed HYPRE package."""
         if "+mpi" not in self.spec:
-            print("Skipping: HYPRE must be installed with +mpi to run tests")
-            return
+            raise SkipTest("Package must be installed with +mpi to run tests")
 
         # Build copied and cached test examples
-        self.run_test(
-            "make",
-            ["HYPRE_DIR={0}".format(self.prefix), "bigint"],
-            purpose="test: building selected examples",
-            work_dir=self._cached_tests_work_dir,
-        )
-
-        # Run the examples built above
-        for exe in ["./ex5big", "./ex15big"]:
-            self.run_test(
-                exe,
-                [],
-                [],
-                installed=False,
-                purpose="test: ensuring {0} runs".format(exe),
-                skip_missing=True,
-                work_dir=self._cached_tests_work_dir,
-            )
+        with test_part(self, "test_hypre_cmake_build", purpose="Building selected examples"):
+            with working_dir(self._cached_tests_work_dir):
+                make = which("make")
+                make("HYPRE_DIR={0}".format(self.prefix), "bigint")
```

Comment:
The convention is to no longer treat the build of tests as separate test parts.  

Rather do the build and run the tests as test_parts all within the working directory.

Reply:
> Rather do the build and run the tests as test_parts all within the working directory.

By this I meant like the `MyLibExample` in the docs at https://spack.readthedocs.io/en/latest/packaging_guide.html#saving-build-and-install-time-files.